### PR TITLE
Login register

### PR DESCRIPTION
**Pull request by kangwork**
Login and Register should work now(Tested several times)

But one issue remains: 

I replaced the usage of props(isLoggedIn) in the Header to the state(isLoggedIn) as it stopped working, sometimes the Logout button is not in sync with the logged-in status (needs to refresh the page)

I did some temp changes as I could not fix the other ways after a while, but will do my best to use the former versions as much as I can later

### CODE REVIEW

File: client/src/layout/Header.tsx [hunk 1]
```diff
@@ -1,19 +1,55 @@
...
+  useEffect(() => {
+    if (!isLoggedIn) {
+      navigate("/login");
+    }
+  }
+  , [isLoggedIn]);
+
+  const onLogout = async () => {
+    try {
+      const response = await fetch(`${API_URL}/accounts/logout/`, {
```
Comment (faizahsayyid) [inline client/src/layout/Header.tsx]: can we use the react query library? you can make these edits in the logout function

File: client/src/pages/Login.tsx [hunk 2]
```diff
@@ -1,19 +1,55 @@
...
 import { Link, useNavigate } from "react-router-dom";
+import { useEffect } from "react";
+import { API_URL } from "../api/config";
 
 export const Login = () => {
-  const loginMutation = useMutation({ mutationFn: login });
   const navigate = useNavigate();
 
+  useEffect(() => {
+    console.log("Entering login page");
```
Comment (faizahsayyid) [inline client/src/pages/Login.tsx]: this logic can be centralized in the root

File: client/src/pages/Login.tsx [hunk 3]
```diff
@@ -1,19 +1,55 @@
...
+      navigate("/");
+    }
+  }, []);
+
   const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     const email = (event.target as HTMLFormElement).email.value;
     const password = (event.target as HTMLFormElement).password.value;
-    console.log(email, password);
-    await loginMutation.mutate({ email, password });
```
Comment (faizahsayyid) [inline client/src/pages/Login.tsx:15]: here as well lets just use react query

File: client/src/pages/Login.tsx (hunk 2 above)
Comment (kangwork) [reply client/src/pages/Login.tsx]: will change this now, thx

File: client/src/layout/Header.tsx (hunk 1 above)
Comment (kangwork) [reply client/src/layout/Header.tsx]: like axios right? I'll do it asap asw, I was just not sure where my login was broken so was changing to what I've learned. Should've changed it back after checking, my bad
Thanks!

File: client/src/pages/Login.tsx (hunk 2 above)
Comment (kangwork) [reply client/src/pages/Login.tsx]: Is it ok if I ask you questions abt the root later in the lab? I changed it and apparently it is not working as it should... 😢 
Screenshot 2024-04-24 at 12 18 41 PM

File: client/src/pages/Login.tsx (hunk 3 above)
Comment (kangwork) [reply client/src/pages/Login.tsx:15]: I tried reverting to that former version, but I think there's some other problem I am not finding... The backend works tho, so it should be a problem in mutation and the exported functions, but have not found it. I like your version better honestly as well, as it simplifies the code, but rn I haven't found a way to debug it. I'll talk to u again in the lab if ur coming!😢

**Pull request merged**